app/models/concerns/has_recent_items.rb

```diff
@@ -27,27 +49,31 @@ def add_recent_item(item)
     def clear_recent_items
       recent_item_links.destroy_all
     end
-
-    def sorted_recent_item_links
-      recent_item_links.order(updated_at: :desc)
-    end
   end
 
   class_methods do
     @recent_item_types = {}
+    @viewable_map = {}
 
     def recent_item_types
       @recent_item_types
     end
 
+    def viewable_map
+      @viewable_map
+    end
+
     # This method should not be converted to 'recent?'
     # It does not query whether this item is recent, it declares a recent item type on the inheriting class
     # rubocop:disable Naming/PredicateName
-    def has_recent(item_type, item_class, name: nil)
+    def has_recent(item_type, item_class, name: nil, viewable_proc: nil)
       association_name = name || "recent_#{item_type}".to_sym
       has_many association_name, through: :recent_item_links, source: :item, source_type: item_class.name
       @recent_item_types = {} if @recent_item_types.nil?
       @recent_item_types[association_name] = item_class
+
+      @viewable_map = {} if @viewable_map.nil?
```

Comment:
Do we need this _and_ line 56?

Reply:
Nope, but we do need `||=` somewhere it seems, so I added it to the method